enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,211 @@
+      /bin/mkdir -p /database && \
+      /bin/opm registry add   -d /database/index.db -b {.BundleImage} && \
+      /bin/opm registry serve -d /database/index.db’
+    ```
+ 1. If pod fails, clean it up, and error out with message of unsupported index image. Also capture the pod's logs for debugging.
+ 1. Create GRPC CatalogSource, pointing to `<pod-name>.<namespace>:50051`
+ 1. If an OperatorGroup doesn't exist, create one and default to `AllNamespaces` as `--install-mode` was not provided
+ 1. Create Subscription
+    * Use default values for `--channel` and `--starting-csv`, and the created `CatalogSource`'s name.
+    * Configure approval to automatic upgrades
+    
+ 1. Operator upgraded into the default namespace defined by KUBECONFIG
+ 1. Verify operator is upgraded (by looking at the CSV version, status, etc.)
+ 1. Log what happened and final state (e.g. operator is running in namespace X, watching namespace(s) Y).
+ 1. Log command to execute to cleanup operator.
+Exit with success (leaving operator running)
+
+#### Complex Example
+
+Let's take an example providing all the required and optional flags in the `upgrade-bundle` command.
+
+```
+operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --channel alpha --namespace memcached-system --install-mode SingleNamespace=default
+```
+
+ 1. Create pod using `--index-image`, set container command (entrypoint) to:
+    ```
+    /bin/bash -c ‘ \
+      /bin/mkdir -p /database && \
+      /bin/opm registry add   -d /database/index.db -b {.BundleImage} && \
+      /bin/opm registry serve -d /database/index.db’
+    ```
+ 1. If pod fails, clean it up, and error out with message of unsupported index image. Also capture the pod's logs for debugging.
+ 1. Create GRPC CatalogSource, pointing to `<pod-name>.<namespace>:50051`
+ 1. Creates an OperatorGroup in and targetting namespace `default`.
+ 1. Create Subscription
+    * Use values from `--channel` and `--starting-csv`, and the created `CatalogSource`'s name.
+    * Configure approval to automatic upgrades
+    
+ 1. Operator upgraded in the namespace `memcached-system`
+ 1. Verify operator is upgraded (by looking at the CSV version, status, etc.)
+ 1. Log what happened and final state (e.g. operator is running in namespace
+    `memcached-system`, watching namespace `default`.
+ 1. Log command to execute to cleanup operator.
+Exit with success (leaving operator running)
+
+#### Prototype Links
+
+ * Prototype for run bundle may be able to be adapted to prototype run bundle-upgrade: https://github.com/joelanford/bundle-runner
+ * Bundle format:
+   https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+
+### Test Plan
+
+  * manual testing with different scenarios 
+  * add unit tests for `internal/olm/operator/operator.go` (with coverage requirements)
+  * investigate adding e2e tests using OLM
```

Comment:
I would like to see this be more of a statement of we will add end to end tests. Are we not e2e testing the olm cli stuff yet?